kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/expressions/ImplicitCastExpression.java

```diff
@@ -108,6 +110,7 @@ ColumnVector eval(ColumnVector input) {
               this.put("integer", Arrays.asList("long", "float", "double"));
               this.put("long", Arrays.asList("float", "double"));
               this.put("float", Arrays.asList("double"));
+              this.put("string", Arrays.asList("string"));
```

Comment:
can you add comments on what it means string is implicitly castable to another string?